core/tests/puppeteer-acceptance-tests/utility-functions/blogPostAdminUtils.js

```diff
@@ -0,0 +1,132 @@
+const { parseHTML } = require("jquery");
+const browser = require("./puppeteer_utils.js");
+const testConstants = require("./testConstants.js");
+
+const blogEditOptions = "e2e-test-blog-post-edit-box";
+const blogTitleInput = "input.e2e-test-blog-post-title-field";
+const blogBodyInput = "div.e2e-test-rte";
+const thumbnailPhotoBox = "e2e-test-photo-clickable";
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+
+
+module.exports = class e2eBlogPostAdmin {
+  page;
+  browserInstance;
+
+  async openBrowser() {
+    this.browserInstance = await new browser();
+    this.page = await (this.browserInstance).initialize();
+  }
+
+  async signInWithEmail(email) {
+    await (this.browserInstance).signInWithEmail(email);
+  }
+
+  async waitForPageToLoad(selector) {
+    await (this.page).waitForSelector(selector);
+  }
+
+  async goto(destination) {
+    await (this.browserInstance).goto(destination);
+  }
+
+  async createDraftBlogPostByTitle(draftBlogPostTitle) {
+    try{
+      await (this.browserInstance).clickOn("span", "NEW POST");
+    } catch {
+      await (this.browserInstance).clickOn("span", " CREATE NEW BLOG POST ");
+    }
+    await (this.browserInstance).type(blogTitleInput, draftBlogPostTitle);
+    await (this.browserInstance).type(blogBodyInput, "blog post test body content");
+    await (this.browserInstance).clickOn("span", " DONE ");
+    await (this.browserInstance).clickOn("span", "SAVE AS DRAFT", 500);
+    
+    await console.log("Successfully created a draft blog post!");
+    await this.browserInstance.goto(blogDashboardUrl);
+
+    // this.page = await this.browserInstance.browser.newPage();
+  }
+
+  async deleteDraftBlogPost() {
+    await (this.browserInstance).clickOn("span", "Delete", 100);
+    await (this.browserInstance).clickOn("button", " Confirm ");
+
+    console.log("draft blog post with given title deleted successfully!");
+  }
+
+  async deleteDraftBlogPostByTitle(draftBlogPostTitle) {
+    let draftBlogPostInfo = {
+      "draftBlogPostTitle": draftBlogPostTitle,
+      "deleteDraftBlogPost": this.deleteDraftBlogPost(),
+    }
+    await (this.page).evaluate(async(draftBlogPostInfo) => {
+      const allDraftBlogPosts = document.getElementsByClassName('blog-dashboard-tile-content');
+      for(let i = 0; i < allDraftBlogPosts.length; i++) {
```

Comment:
Make sure to throw an error if you can't find the blog post in question, or if there is > 1 blog post with that name.

Reply:
@Shivkant-Chauhan You didn't reply to this comment, I think.